src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,181 @@
+
+    final static String SERVICE_ACCOUNT_PASSWORD_MESSAGE = "A password cannot be provided for a service account. Failed to register service account: ";
+
+    final static String SERVICE_ACCOUNT_HASH_MESSAGE = "A password hash cannot be provided for service account. Failed to register service account: ";
+
+    final static String NO_ACCOUNT_NAME_MESSAGE = "No account name was specified in the request.";
+    private static CType getConfigName() {
+        return CType.INTERNALUSERS;
+    }
+
+    static final List<String> RESTRICTED_FROM_USERNAME = ImmutableList.of(
+            ":" // Not allowed in basic auth, see https://stackoverflow.com/a/33391003/533057
+    );
+
+    @Inject
+    public UserService(
+            ClusterService clusterService,
+            ConfigurationRepository configurationRepository,
+            Settings settings,
+            Client client
+    ) {
+        this.clusterService = clusterService;
+        this.configurationRepository = configurationRepository;
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
+                ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        this.client = client;
+    }
+
+    /**
+     * Load data for a given CType
+     * @param config CType whose data is to be loaded in-memory
+     * @return configuration loaded with given CType data
+     */
+    protected static final SecurityDynamicConfiguration<?> load(final CType config, boolean logComplianceEvent) {
+        SecurityDynamicConfiguration<?> loaded = configurationRepository.getConfigurationsFromIndex(Collections.singleton(config), logComplianceEvent).get(config).deepClone();
+        return DynamicConfigFactory.addStatics(loaded);
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param contentAsNode An object node of different account configurations.
+     * @return InternalUserConfiguration with the new/updated user
+     * @throws UserServiceException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(ObjectNode contentAsNode) throws IOException {
+
+
+        SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getConfigName(), false);
+        String accountName = securityJsonNode.get("name").asString();
+
+        if (accountName == null || accountName.length() == 0) { // Fail if field is present but empty
+            throw new UserServiceException(NO_ACCOUNT_NAME_MESSAGE);
+        }
+
+        if (!securityJsonNode.get("attributes").get("owner").isNull() && !securityJsonNode.get("attributes").get("owner").equals(accountName)) { // If this is a service account
+            verifyServiceAccount(securityJsonNode, accountName);
+            String password = generatePassword();
```

Comment:
why not use `securityJsonNode.put('name')...` here to avoid duplicate declaration on line 115?
Having said that, I know that `put()` doesn't exist in SecurityJsonNode for some unknown reason (probably to make it a read-only node). Do you think creating it might be worthwhile?

Reply:
See above. I can remove use of the contentAsNode pattern by adding methods to the SecurityJsonNode object. I was not sure which implementation would be preferred. 

I am interpreting this as you would prefer that change so I will go that route.

Reply:
EDIT: I looked into this, looks like the existing SecurityJsonNode is based off of Jackson Databind's JSONNode class. That class is not constructed for removal or addition of elements. I am going to stick with this implementation for the time being unless that is a blocker to this PR.

Reply:
EDIT: Hi @DarshitChanpura Unless it is going to block this PR, I am going to suggest we leave it as is. I know it is not the most convenient solution but the securityJsonNode class has 40+ usages and was not intended to be directly modifiable. I have been working on modifying it but it requires changing the class from creating final instances of itself to creating mutable versions. I suspect a part of the reason the class was created how it is, is so that there would be an immutable version of a JSON node for implementations. Let me know if this will be a blocker and I can look more into it but I am going to suggest we leave this as is otherwise. 

P.S. I did make the other changes you suggested.

Reply:
Alright, we can leave it as is. Thank you for taking time and going through SecurityJsonNode 😃. We can cover this in a separate PR in future